Modules/Daq/src/DaqTask.cxx

```diff
@@ -15,140 +15,249 @@
-  mIds->GetXaxis()->SetNdivisions(-503);
-  mIds->GetXaxis()->SetTimeFormat("%Y-%m-%d %H:%M:%S");
-  mIds->GetXaxis()->SetTimeOffset(0, "gmt");
-  mIds->SetMarkerStyle(20);
-  mIds->GetXaxis()->SetLabelSize(0.02);
-  mIds->GetYaxis()->SetLabelSize(0.02);
-  getObjectsManager()->startPublishing(mIds);
+  ILOG(Info, Support) << "Initializiation of DaqTask" << ENDM;
+
+  // General plots, related mostly to the payload size (InputRecord, Inputs) and the numbers of RDHs and Inputs in an InputRecord.
+  mInputRecordPayloadSize = new TH1F("inputRecordSize", "Total payload size per InputRecord;bytes", 128, 0, 2047);
+  mInputRecordPayloadSize->SetCanExtend(TH1::kXaxis);
+  getObjectsManager()->startPublishing(mInputRecordPayloadSize);
+  mNumberInputs = new TH1F("numberInputs", "Number of inputs per InputRecords", 100, 1, 100);
+  getObjectsManager()->startPublishing(mNumberInputs);
+  mInputSize = new TH1F("payloadSizeInputs", "Payload size of the inputs;bytes", 128, 0, 2047);
+  mInputSize->SetCanExtend(TH1::kXaxis);
+  getObjectsManager()->startPublishing(mInputSize);
+  mNumberRDHs = new TH1F("numberRDHs", "Number of RDHs per InputRecord", 100, 1, 100);
+  mNumberRDHs->SetCanExtend(TH1::kXaxis);
+  getObjectsManager()->startPublishing(mNumberRDHs);
+
+  // initialize a map for the subsystems (id, name)
+  for (int i = DAQID::MINDAQ; i < DAQID::MAXDAQ + 1; i++) {
+    DataOrigin origin = DAQID::DAQtoO2(i);
+    if (origin != gDataOriginInvalid) {
+      mSystems[i] = origin.str;
+    }
+  }
+  mSystems[DAQID::INVALID] = "UNKNOWN"; // to store RDH info for unknown detectors
+
+  // subsystems plots: distribution of rdh size, distribution of the sum of rdh in each message.
+  for (const auto& system : mSystems) {
+    string name = system.second + "/sumRdhSizesPerInputRecord";
+    string title = "Sum of RDH sizes per InputRecord for " + system.second + ";bytes";
+    mSubSystemsTotalSizes[system.first] = new TH1F(name.c_str(), title.c_str(), 128, 0, 2047);
+    mSubSystemsTotalSizes[system.first]->SetCanExtend(TH1::kXaxis);
+    getObjectsManager()->startPublishing(mSubSystemsTotalSizes[system.first]);
+
+    name = system.second + "/RdhSizes";
+    title = "RDH sizes for " + system.second + ";bytes";
+    mSubSystemsRdhSizes[system.first] = new TH1F(name.c_str(), title.c_str(), 128, 0, 2047);
+    mSubSystemsRdhSizes[system.first]->SetCanExtend(TH1::kXaxis);
+    getObjectsManager()->startPublishing(mSubSystemsRdhSizes[system.first]);
+  }
 }
 
-void DaqTask::startOfActivity(Activity& /*activity*/)
+void DaqTask::startOfActivity(Activity& activity)
 {
-  QcInfoLogger::GetInstance() << "startOfActivity" << AliceO2::InfoLogger::InfoLogger::endm;
-  mPayloadSize->Reset();
-  mNumberSubblocks->Reset();
-  mSubPayloadSize->Reset();
-  mIds->Set(0);
-  mNPoints = 0;
+  ILOG(Info, Support) << "startOfActivity: " << activity.mId << ENDM;
+  reset(); // TODO make sure this is what we want actually, but most probably it is
```

Comment:
It shouldn't have any real influence, because newly created histograms shouldn't contain any data and during stop()  we do:
```
  endOfActivity();
  mTask->reset();
```